lua/zk/lsp.lua

```diff
@@ -18,16 +18,7 @@ function M.external_client()
     active_clients = vim.lsp.get_active_clients({ name = client_name })
   end
 
-  if active_clients == {} then
-    return nil
-  end
-
-  -- return first lsp server that is actually in use
-  for _, v in ipairs(active_clients) do
```

Comment:
Check if client attached any buffer is not the way to tell if it were in use.

Reply:
again. this wasn't the agreed upon solution from #173. Simply returning the first active client may introduce a bug, as we want to return _the first client that is actually attached to a buffer_. In #173 you were also implementing a user side solution (your own solution) to disable `auto_attach` in your config. 

If that client was the first in the table, and it was returned then we would be attempting to use a client that wasn't actually attached to the buffer.. Furthermore, there can be active clients attached to different buffers. So returning a client that isn't actually attached to the buffer in question would also be a potential bug enabled by removing this `if` check.

I'm happy for these prs, but for the sake of time, I'm going to push the rest of the changes myself and merge.

Also, I've seen in some of your other PRs that you're using chatGPT. The backwards and forwards we've had with unexpected code changes between comments / prs reminds me a lot of a chatGPT workflow, and has made this all take a lot more time, confused me and made me doubt my decision making. I hope the changes you've been suggesting were not blind suggestions from chatGPT...

Reply:
I did all the code myself. If my code was bad, I apologize.

> So returning a client that isn't actually attached to the buffer in question would also be a potential bug

I don't understand why a client had no buffer attached could be a potential bug, soon it will attach one as they had well configured, and zk-nvim shouldn't try to create an additional one.